luigi/scheduler.py

```diff
@@ -96,12 +95,15 @@ def rpc_method(**request_args):
     def _rpc_method(fn):
         # If request args are passed, return this function again for use as
         # the decorator function with the request args attached.
-        fn_args = six.getargspec(fn)
-
-        assert not fn_args.varargs
-        assert fn_args.args[0] == 'self'
-        all_args = fn_args.args[1:]
-        defaults = dict(zip(reversed(all_args), reversed(fn_args.defaults or ())))
+        args, varargs, varkw, defaults, kwonlyargs, kwonlydefaults, ann = inspect.getfullargspec(fn)
+        if kwonlyargs or ann:
+            raise ValueError("Function has keyword-only parameters or annotations"
```

Comment:
Not due to this PR, but I'm not sure what this error tries to tell. Here we already use `getfullargspec`. Do I miss anything?

Reply:
@honnix 
I think that's related to the fact, this is going on inside of function `getargspec` which is not supposed to deal with annotations and keyword-only parameters
so this exception seems to be some sort of code-consistency among python version
but I'm still unsure.
I think we can remove this raising